pkg/apis/psmdb/v1/psmdb_types.go

```diff
@@ -262,6 +262,8 @@ const (
 	AppStatePaused   AppState = "paused"
 	AppStateReady    AppState = "ready"
 	AppStateError    AppState = "error"
+
+	AppStateSharding AppState = "sharding"
```

Comment:
i don't think this should be type `AppState` but I know that we use a custom condition struct rather than using the one exported from `meta/v1`. we need to start using it in near future